app/src/main/kotlin/compiler/intermediate/generators/DefaultFunctionDetailsGenerator.kt

```diff
@@ -67,7 +68,7 @@ data class DefaultFunctionDetailsGenerator(
         cfgBuilder.addLinksFromAllFinalRoots(
             CFGLinkType.UNCONDITIONAL,
             IFTNode.StackPush(IFTNode.RegisterRead(Register.RBP))
-        )
+        ) // disaligns stack
```

Comment:
Actually, this PUSH aligns the stack to 16B. SysV64 convention says that right after CALL, RSP % 16 = 8, so the push makes it aligned.

Reply:
Consequently, I am not sure whether the  whole program is correct as:
At this point (after PUSH) stack is aligned.
After local variable allocation stack is 8 modulo 16.
After pushing the arguments (while calling some other function) the stack is still 8 modulo 16.
After CALL, stack is aligned.
After PUSH, stack is not aligned.

Reply:
Indeed, it seems that right now CALLs are at 8 mod 16. Changing the remainder to 0 at line 90 should fix the problem.

Reply:
Thanks for spotting, fixed